File: JetPackStudy/viewmodel/src/main/java/com/example/viewmodel/MainActivity.java
package com.example.viewmodel;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    protected TextView textView;
    protected MyViewModel myViewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textView = findViewById(R.id.textView);

        myViewModel = new ViewModelProvider(this,new ViewModelProvider.AndroidViewModelFactory(getApplication())).get(MyViewModel.class);
        textView.setText(String.valueOf(myViewModel.num));
    }

    public void plusNum(View view) {
        textView.setText(String.valueOf(++myViewModel.num));
    }
}

File: JetPackOfKotlin/app/src/main/java/com/example/jetpackofkotlin/MainActivity.kt
package com.example.jetpackofkotlin

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.example.jetpackofkotlin.MyViewModel.MyViewModel
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var num:Int = 0
    lateinit var viewModel: MyViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        viewModel = ViewModelProvider(this)[MyViewModel::class.java]
        viewModel.num.observe(this, Observer {
            tv1.text = it.toString()
        })
        tv1.text = "0"
    }

    fun increase(view: View) {
        viewModel.modifyNum(1)
    }
    fun decrease(view: View) {
        viewModel.modifyNum(-1)
    }

}